### DISCUSSION TITLE
Grouping Day and Time as one exit in Time Switch Node

### DISCUSSION THREAD

**Discussion by ddiwani** [jensrossbach/node-red-contrib-chronos#87] [Ideas & Requests]
Would it be possible to group both day and time as a single exit using a time switch node? Currently I can achieve the the same but would have to use two time nodes...one the evaluates the days and second node for time. 

Here is the used case. My business Open Days/Hours are as follows...
Monday - Friday from 8am - 17pm
Saturday - 10am - 4pm.

When above is true - then send flow to normal business hours flow. Otherwise, send flow to after hour flow.

It would nice if the time switch node can add the following enhancements (using the single node)
Use And Operator to group the first set
((Monday - Friday) AND (8am - 17pm))
Combine with an OR operator for below
((Saturday) AND (10am - 4pm))

As one exit  to cover the during business days/hours operation

Otherwise which already exists to cover the after hours operation.

**Comment by jensrossbach** [comment]
Hi,

what is wrong with your solution using two time switch nodes? In my opinion this sounds pretty perfect. To be honest I believe that integrating what you want into the node's configuration is not possible without cluttering the UI and make it too complex and hard to understand. I would like to keep the nodes as lean as possible and chaining the nodes in order to realize certain boolean expressions was definitely foreseen in the design of the nodes.

**Reply by jensrossbach** [reply]
And by the way, if you really only want to use a single node, another solution could be to use the _expression_ condition. The following JSONata expression could do the job:

```
($matchesWeekdays(
   $baseTime,
   {
       "monday": true,
       "tuesday": true,
       "wednesday": true,
       "thursday": true,
       "friday": true
   }) and $isBetween(
       $baseTime,
       "time",
       "8:00",
       0,
       false,
       "time",
       "17:00",
       0,
       false
   )
) or (
   $matchesWeekdays($baseTime, {"saturday": true}) and $isBetween(
       $baseTime,
       "time",
       "10:00",
       0,
       false,
       "time",
       "16:00",
       0,
       false
   )
)
```

**Reply by ddiwani** [reply]
I may not be needing this anymore as the other since the solution provided below is what I have been looking to do. Sorry for the late reply.

**Reply by jensrossbach** [reply]
No problem, but be aware that the other solution still requires two (filter) nodes, while above solution can be done with a single (switch) node. However, if you don't need the 'else' path, or want to configure the conditions only via graphical UI, the other solution might fit better.

**Reply by ddiwani** [reply]
Thanks. One other question, is possible to add custom date so I can evaluate this as a holiday. So, say I have a custom holiday on August 12, 2022. How can I incorporate that into the time node?

**Reply by jensrossbach** [reply]
Wouldn't a condition with the 'days' operator do what you want? Like this:
![image](https://user-images.githubusercontent.com/5118924/179349171-766eff42-bd1d-4e9f-a234-e421c18657cf.png)
Or did I understand your requirement wrongly?

**Comment by crxporter** [comment]
I did this same kind of thing just this morning, I used a JSONata rule for the evaluation field on a time filter node like:

```
{($condition[0] and $condition[1]) or ($condition[2] and $condition[3])}
```

Then I put filter 0 for my weekday hours, filter 1 for weekdays check box, filter 2 for the weekend hours, and filter 3 for weekend days checked. 

Works great! I'll post a screenshot in just a minute, I have to go find my computer.

**Reply by crxporter** [reply]
<img width="834" alt="image" src="https://user-images.githubusercontent.com/38265886/169593551-48712002-1c46-4011-8886-bf20e233e57c.png">

Like was stated above, if you want both "between" and "outside" of those hours you'd need another filter for the opposite case. I was just wanting to pass messages if it's daytime.

**Reply by ddiwani** [reply]
This is exactly what I have been look for, thanks a bunch!

**Comment by ddiwani** [comment]
Anyone has a way to add a custom date and time condition? This is so I can evaulate a custom holiday schedule. Unfortunately, I have not been able to find a node that I can group date and time and even a title for the holiday without using a function/java script. Please advise.

**Reply by jensrossbach** [reply]
Does my [response above](https://github.com/jensrossbach/node-red-contrib-chronos/discussions/87#discussioncomment-3161357) answer your question?
